### DISCUSSION TITLE
Views for KEK-Extension

### DISCUSSION THREAD

**Discussion by urskaufmann** [QGEP/QGEP#703] [General]
The views vw_qgep_maintenance-event and vw_qgep_damage and the way they are configured in the qgep-project is not very useful. We should probably have a good solution for user, who use the maintenance_events, but without KEK, and users , who use KEK. A KEK-user wants to use the damage-table and therefor has to create examinations in table examination (not only a maintenance-event with kind = examination).
They way it is configured now, I can create a maintenace-event with kind = examination, but I can not create a examination-record and therefor I can not create damages that are connected.
Idea: Non KEK-user have only the od_maintenance_event layer and no damage layer. For KEK-user there must be a simple way to create examination-records and the layer is called vw_qgep_maintenance_event_KEK

**Comment by sjib** [comment]
There are actually more use cases we should consider:
a) maintenance_event (Erhaltungsereignis), maintenance.kind can be any.
 (the case, when maintenance is a dataset not within a subclass will become a subclass in release 2020 called maintenance (Unterhalt), to make maintenance_event an abstract class and deal with it in the same way for all datasets (which makes it simpler). 
b) bio_ecol_assessment (Biol_oekol_Gesamtbeurteilung) from the extension Stammkarte), subclass of maintenance_event, maintenance_event.kind = examination
c) damage_channel (Kanalschaden) and damage_manhole (Normschachtschaden) (from VSA-KEK), sub_sub_classes of maintenance_event, maintenance.kind = examination, subclasses of damage (Schaden). Damages should always be one of the two: 
d) measure (Massnahme) – which can be linked via a maintenance_event to one or many wastewater_structures (new relation in release 2020)

![klassendiagramm_2020_erhaltungsereignis_massnahme_ausschnitt](https://user-images.githubusercontent.com/1265865/138843163-760a0fab-1f46-4036-be4a-3830efcb9a1d.jpg)
_Class Diagramm VSA-DSS 2020_

So I agree with the idea above and would suggest to also add a third option (vw_biol_ecol_assessment) for adding biological ecological assessments to discharge_points and when we have the 2020 release also a forth option to add and link measures.
